### DISCUSSION TITLE
Does CopyQ store all content in RAM unless instructed otherwise?

### DISCUSSION THREAD

**Discussion by Gitoffthelawn** [hluk/CopyQ#2102] [Q&A]
When using CopyQ with a single `&clipboard` tab, does CopyQ store all content in RAM unless explicitly instructed otherwise?

(BTW, it seems to do this, which in my opinion, is a good thing.)

**Comment by hluk** [comment]
Yes, CopyQ loads all data for all the items in the current tab (or recently opened tab) to memory (until it expires).

**Reply by Gitoffthelawn** [reply]
> Yes, CopyQ loads all data for all the items in the current tab (or recently opened tab) to memory (until it expires).

Thanks Lukas.  And just to clarify, CopyQ never saves them to disk unless the user explicitly instructs it to do so.  Is that correct?

**Reply by hluk** [reply]
All tab data are automatically saved to disk by default. Otherwise the tab content would be lost.

**Reply by henri** [reply]
Within the settings -> tabs -> <clipboard-name> there is an checkbox "Save items". If this is unchecked then is it true that data will only be stored in RAM and not sent to disk (unless RAM is swapped to disk by the operating system).

![image](https://github.com/user-attachments/assets/7aaf654a-7a56-4750-87d9-fe9fc1a6619c)

**Comment by henri** [comment]
See this issue which has been answered : https://github.com/hluk/CopyQ/discussions/1656
